lib/buffer.js

```diff
@@ -272,19 +272,66 @@ Buffer.concat = function(list, length) {
 };
 
 
+function base64ByteLength(str) {
+  var len = str.length;
+  var bytes = len;
+
+  // Handle padding
+  if (str[len - 1] === '=')
+    bytes--;
+  if (len > 2 && str[len - 2] === '=')
+    bytes--;
+
+  // Base64 ratio: 3/4
+  bytes = (bytes / 4) * 3;
```

Comment:
As it is, the new base64 is *much* faster right now, so I think any difference this would make would be magnitudes smaller anyways.